models.rdf

```diff
@@ -0,0 +1,93 @@
+      <rdfs:label xml:lang="en">Collection</rdfs:label>
+      <rdfs:comment xml:lang="en">
+        A Collection is a group of resources. Collections have descriptive metadata, access metadata,
+        and may links to works and/or collections. By default, member works and collections are an
+        unordered set, but can be ordered using the ORE Proxy class.
+      </rdfs:comment>
+      <rdfs:subClassOf rdf:resource="http://www.openarchives.org/ore/terms/Aggregation"/>
+      <rdfs:isDefinedBy rdf:resource="http://pcdm.org/models#"/>
+    </rdfs:Class>
+
+    <rdfs:Class rdf:about="http://pcdm.org/models#Object">
+      <rdfs:label xml:lang="en">Object</rdfs:label>
+      <rdfs:comment xml:lang="en">
+        An Object is an intellectual entity, sometimes called a "work", "digital object", etc.
+        Objects have descriptive metadata, access metadata, may contain files and other Objects as
+        member "components". Each level of a work is therefore represented by an Object instance,
+        and is capable of standing on its own, being linked to from Collections and other Objects.
+        Member Objects can be ordered using the ORE Proxy class.
+      </rdfs:comment>
+      <rdfs:subClassOf rdf:resource="http://www.openarchives.org/ore/terms/Aggregation"/>
+      <rdfs:isDefinedBy rdf:resource="http://pcdm.org/models#"/>
+    </rdfs:Class>
+   
+    <rdfs:Class rdf:about="http://pcdm.org/models#File">
+      <rdfs:label xml:lang="en">File</rdfs:label>
+      <rdfs:comment xml:lang="en">
+        A File is a sequence of binary data and is described by some accompanying metadata.
+        The metadata typically includes at least basic technical metadata (size, content type,
+        modification date, etc.), but can also include properties related to preservation,
+        digitization process, provenance, etc. Files MUST be contained by exactly one Object.
+      </rdfs:comment>
+      <rdfs:isDefinedBy rdf:resource="http://pcdm.org/models#"/>
+    </rdfs:Class>
+
+    <rdf:Property rdf:about="http://pcdm.org/models#hasFile">
+      <rdfs:label xml:lang="en">has file</rdfs:label>
+      <rdfs:comment xml:lang="en">Links to a File contained by this Object.</rdfs:comment>
+      <rdfs:domain rdf:resource="http://pcdm.org/models#Object"/>
+      <rdfs:range rdf:resource="http://pcdm.org/models#File"/>
+      <rdfs:subPropertyOf rdf:resource="http://www.w3.org/ns/ldp#contains"/>
+      <rdfs:isDefinedBy rdf:resource="http://pcdm.org/models#"/>
+    </rdf:Property>
+
+    <rdf:Property rdf:about="http://pcdm.org/models#hasMember">
+      <rdfs:label xml:lang="en">has member</rdfs:label>
+      <rdfs:comment xml:lang="en">Links to a related Object. Typically used to link to component parts, such as a book linking to a page.</rdfs:comment>
+      <rdfs:domain rdf:resource="http://www.openarchives.org/ore/terms/Aggregation"/>
+      <rdfs:range rdf:resource="http://www.openarchives.org/ore/terms/Aggregation"/>
```

Comment:
Also, remember that domain and range do not allow validation. They are
constructs for inferencing engines.
It's always legitimate in RDF to make any assertion you want, regardless of
domain and range.   The fewer inferencing requirements we have, the more
reusable our properties and relationships will be.

So I'm happy with domain and range of Aggregation and not creating abstract
intermediary classes, and reusing hasMember for collections and objects,

Rob

On Mon, Mar 16, 2015 at 10:15 AM, Jon Stroop <notifications@github.com>
wrote:

> In models.rdf
> <https://github.com/duraspace/pcdm/pull/1#discussion_r26500067>:
>
> > +    </rdfs:Class>
> > +
> > +    <rdf:Property rdf:about="http://pcdm.org/models#hasFile">
> > +      <rdfs:label xml:lang="en">has file</rdfs:label>
> > +      <rdfs:comment xml:lang="en">Links to a File contained by this Object.</rdfs:comment>
> > +      <rdfs:domain rdf:resource="http://pcdm.org/models#Object"/>
> > +      <rdfs:range rdf:resource="http://pcdm.org/models#File"/>
> > +      <rdfs:subPropertyOf rdf:resource="http://www.w3.org/ns/ldp#contains"/>
> > +      <rdfs:isDefinedBy rdf:resource="http://pcdm.org/models#"/>
> > +    </rdf:Property>
> > +
> > +    <rdf:Property rdf:about="http://pcdm.org/models#hasMember">
> > +      <rdfs:label xml:lang="en">has member</rdfs:label>
> > +      <rdfs:comment xml:lang="en">Links to a related Object. Typically used to link to component parts, such as a book linking to a page.</rdfs:comment>
> > +      <rdfs:domain rdf:resource="http://www.openarchives.org/ore/terms/Aggregation"/>
> > +      <rdfs:range rdf:resource="http://www.openarchives.org/ore/terms/Aggregation"/>
>
> That does seem simplest, and correct alternatives all seem too complex.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duraspace/pcdm/pull/1/files#r26500067>.
>



-- 
Rob Sanderson
Information Standards Advocate
Digital Library Systems and Services
Stanford, CA 94305